crypto/multisig/src/lib.rs

```diff
@@ -146,35 +187,47 @@ impl<P: Default + PublicKey + Message> SignerSet<P> {
         // While the verification code below should be immune to duplicate signers or
```

Comment:
:x: I don't think this properly handles the case of a common signer.
```rust
        if multi_sig.signatures.len() < self.threshold as usize
```
What if this is the root set and the threshold is 2, but there is only one signature.  IFF the signature happens to be a common one used in 2 of the subsets and is sufficient to fulfill the subset needs then 2 subsets will be met with only one signature.

Reply:
Good catch, I will add a test case for this.